misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/FakeJobQueueTest.kt

```diff
@@ -209,22 +425,34 @@ private enum class Color {
 
 private class ColorException : Throwable()
 
-private data class ExampleJob(val color: Color, val message: String)
+private data class ExampleJob(
+  val color: Color,
+  val message: String,
+  val hint: ExampleJobHint? = null
+)
+
+private enum class ExampleJobHint {
```

Comment:
I've added also THROW_ONCE to support tests in `FakeTransactionalJobQueueTest`. I saw transactional queue is deprecated, so I anticipate all tests might be moved here eventually.